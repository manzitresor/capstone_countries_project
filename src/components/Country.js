/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

function Country({ country }) {
  const { name, region, flags } = country;
  return (
    <>
      <div className="card">
        <NavLink to={`/details/${name.official}`} className="navLink">
          <img src={flags.png} alt="country" />
          <h1 className="country-heading-one">
            Country Name:
            {' '}
            {name.official}
          </h1>
          <h4>
            Region:
            {' '}
            {region}
          </h4>
        </NavLink>
      </div>
    </>
  );
}

export default Country;

Country.propTypes = {
  country: PropTypes.shape({
    name: PropTypes.shape({ official: PropTypes.string.isRequired }).isRequired,
    flags: PropTypes.shape({ png: PropTypes.string.isRequired }).isRequired,
    region: PropTypes.string.isRequired,
  }).isRequired,
};
