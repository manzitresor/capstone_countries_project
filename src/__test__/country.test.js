import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter to mock react-router-dom
import Country from '../components/Country';
import '@testing-library/jest-dom/extend-expect';

describe('Country component test', () => {
  const mockCountry = {
    name: 'Test Country',
    region: 'Test Region',
    flags: 'test_image.png',
  };

  it('should render Country component correctly', () => {
    const { getByText, getByAltText, getByTestId } = render(
      // Wrap the component in MemoryRouter to mock react-router-dom
      <MemoryRouter>
        <Country country={mockCountry} />
      </MemoryRouter>,
    );

    // Check if the country name is rendered correctly
    const countryName = getByText('Test Country');
    expect(countryName).toBeInTheDocument();

    // Check if the region is rendered correctly
    const regionName = getByText('Region: Test Region');
    expect(regionName).toBeInTheDocument();

    // Check if the image alt text is correct
    const countryImage = getByAltText('country');
    expect(countryImage).toBeInTheDocument();

    // Check if NavLink is present and has the correct destination
    const navLink = getByTestId('navLink');
    expect(navLink).toBeInTheDocument();
    expect(navLink).toHaveAttribute('href', '/details/Test Country');
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Country country={mockCountry} />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
