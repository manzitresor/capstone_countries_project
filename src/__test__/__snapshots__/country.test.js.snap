// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component test should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <a
      class="navLink"
      data-testid="navLink"
      href="/details/Test Country"
    >
      <div
        class="image"
      >
        <img
          alt="country"
          src="test_image.png"
        />
      </div>
      <h1
        class="country-heading-one"
      >
        Test Country
      </h1>
      <h4
        class="country-heading-two"
      >
        Region: Test Region
      </h4>
    </a>
  </div>
</DocumentFragment>
`;
